fix(billing): Fix on-demand capitalization

dashed opened the pull request:
Closes https://linear.app/getsentry/issue/BIL-1646/legacy-tiers-correct-capitalization-in-billing-cycle-card

dashed committed:
fix(billing): Fix on-demand capitalization

isabellaenriquez commented on static/gsApp/views/amCheckout/steps/setSpendLimit.tsx:
```diff
@@ -55,7 +56,7 @@ function SetSpendCap({
         header={
           <StepHeader
-            title={t('Set your %s limit', activePlan.budgetTerm)}
+            title={t('Set your %s limit', displayBudgetName(activePlan, {title: true}))}
```
this shouldn't be title case; it was requested we keep steps sentence-case

isabellaenriquez commented on static/gsApp/views/amCheckout/cart.tsx:
```diff
@@ -238,13 +239,16 @@ function ItemsSummary({activePlan, formData}: ItemsSummaryProps) {
+                                budgetTerm:
+                                  activePlan.budgetTerm === 'pay-as-you-go'
+                                    ? 'PAYG'
+                                    : displayBudgetName(activePlan, {title: true}),
```
can we also keep this sentence case for OD in the whole cart? i think it should only title case if it's at the start of the sentence

isabellaenriquez reviewed (commented):
thanks!

dashed committed:
fix(billing): Use sentence case for on-demand budget terms

Addresses PR feedback to keep budget term capitalization in sentence case
for step titles and cart text, except when at the beginning of a sentence.

dashed committed:
fix
